File: rezervacije/app/services/audit_service.py
from app.repositories import audit_repository


def log_reservation_audit(db, action: str, reservation_id: int | None, actor: str, details: dict):
    return audit_repository.create_audit_log(
        db,
        {
            "service_name": "reservations-service",
            "entity_type": "reservation",
            "entity_id": reservation_id,
            "action": action,
            "actor": actor or "system",
            "details": details or {},
        },
    )


def get_audit_logs(db, action: str | None = None, limit: int = 50):
    return audit_repository.get_audit_logs(db, action=action, limit=limit)

File: inventar/src/services/auditService.js
const auditRepository = require("../repositories/auditRepository");

async function logInventoryAudit(action, item, details = {}, actor = "system") {
    return auditRepository.createAuditLog({
        service_name: "inventory-service",
        entity_type: "item",
        entity_id: item?.id ?? details?.item_id ?? null,
        action,
        actor,
        details,
    });
}

async function getAuditLogs(filters) {
    return auditRepository.getAuditLogs(filters);
}

module.exports = {
    logInventoryAudit,
    getAuditLogs,
};
